src/app/pages/storage2/modules/devices/components/devices/devices.component.ts

```diff
@@ -104,8 +104,22 @@ export class DevicesComponent implements OnInit {
           tap((disks) => {
             this.diskDictionary = _.keyBy(disks, (disk) => disk.devname);
             this.topology = pools[0].topology;
-            this.treeControl.dataNodes = this.topology.data;
-            this.createDataSource(this.topology.data);
+            const data = { children: this.topology.data, type: 'Data VDEVs', guid: 'data' } as unknown as VDev;
```

Comment:
Needs proper typing. This is not really a VDev, but a separate thing in tree table, so we need to introduce another type and replace VDev in `NestedTreeControl`.

Also method starts to get fat, time to extract some logic to a separate method, e.g. `createDataSource`.

Reply:
Probably type should be an enum